src/app/services/form-element-control.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { FormControl, FormGroup, ValidatorFn, Validators } from '@angular/forms';
+import { FormElementBase } from '../shared/dynamic-form/types/form-element-base';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FormElementControlService {
+  toFormGroup = (formElements: FormElementBase<string>[]) => {
+    const group: { [key: string]: FormControl } = {};
+    let validations: ValidatorFn[] = [];
+    // TODO : Extract the validation logic in separate ts
```

Comment:
I don't think we need todo for this right now - but I would extract constants for the magic numbers below:
```js
  if (formElement.validation?.includes('age')) {
        validations.push(Validators.max(100));
      }
      if (formElement.validation?.includes('password')) {
        validations.push(Validators.minLength(8));
      }
```

Reply:
Done.